### PR TITLE
fix source root locations.

### PR DESCRIPTION
**Pull request by pbogdan**
These have changed following commit https://github.com/NixOS/nixpkgs/commit/c3255fe8ec326d2c8fe9462d49ed83aa64d3e68f.

See also:

- https://github.com/NixOS/nixpkgs/pull/31004
- https://github.com/NixOS/nixpkgs/pull/31106

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/audio/puddletag/default.nix [hunk 1]
```diff
@@ -11,7 +11,9 @@ python2Packages.buildPythonApplication rec {
     sha256 = "1g6wa91awy17z5b704yi9kfynnvfm9lkrvpfvwccscr1h8s3qmiz";
   };
 
-  sourceRoot = "source/source";
+  setSourceRoot = ''
+    sourceRoot=$(echo */source)
+  '';
```
Comment (pbogdan) [inline pkgs/applications/audio/puddletag/default.nix]: It's been suggested not to use "source" directly - https://github.com/NixOS/nixpkgs/pull/31106#issuecomment-341266441 - hence I changed it as per the comment.

File: pkgs/tools/cd-dvd/unetbootin/default.nix [hunk 2]
```diff
@@ -12,7 +12,9 @@ stdenv.mkDerivation rec {
     sha256 = "18bbcrjk6ladr46kl3dvqz5pq2xcv4nnwmajqllb4sl3k1xqsngy";
   };
 
-  sourceRoot = "source/src/unetbootin";
+  setSourceRoot = ''
+    sourceRoot=$(echo */src/unetbootin)
+  '';
```
Comment (pbogdan) [inline pkgs/tools/cd-dvd/unetbootin/default.nix]: It's been suggested not to use "source" directly - https://github.com/NixOS/nixpkgs/pull/31106#issuecomment-341266441 - hence I changed it as per the comment.

File: pkgs/tools/misc/cloc/default.nix [hunk 3]
```diff
@@ -11,7 +11,9 @@ stdenv.mkDerivation rec {
     sha256 = "1ihma4f6f92jp1mvzr4rjrgyh9m5wzrlxngaxfn7g0a8r2kyi65b";
   };
 
-  sourceRoot = "source/Unix";
+  setSourceRoot = ''
+    sourceRoot=$(echo */Unix)
+  '';
```
Comment (pbogdan) [inline pkgs/tools/misc/cloc/default.nix]: It's been suggested not to use "source" directly - https://github.com/NixOS/nixpkgs/pull/31106#issuecomment-341266441 - hence I changed it as per the comment.

File: pkgs/development/compilers/dmd/2.067.1.nix [hunk 4]
```diff
@@ -17,18 +17,21 @@ stdenv.mkDerivation rec {
...
     sha256 = "1n2qfw9kmnql0fk2nxikispqs7vh85nhvyyr00fk227n9lgnqf02";
+    name = "druntime-v${version}-src";
   })
   (fetchFromGitHub {
     owner = "dlang";
     repo = "phobos";
     rev = "v${version}";
     sha256 = "0fywgds9xvjcgnqxmpwr67p3wi2m535619pvj159cgwv5y0nr3p1";
+    name = "phobos-v${version}-src";
   })
```
Comment (pbogdan) [inline pkgs/development/compilers/dmd/2.067.1.nix]: Unsure if this is proper way to deal with derivation with multiple sources.

File: pkgs/development/compilers/dmd/default.nix [hunk 5]
```diff
@@ -14,18 +14,21 @@ stdenv.mkDerivation rec {
...
     sha256 = "0idn2v1lmp7hl637g3i7pdfj9mjk4sclkz4cm77nl8873k2fhk8j";
+    name = "druntime-v${version}-src";
   })
   (fetchFromGitHub {
     owner = "dlang";
     repo = "phobos";
     rev = "v${version}";
     sha256 = "1a7q5fd15yspgs5plxgx54jyrcwgzlyw3rahmz04jd2s5h56dj04";
+    name = "phobos-v${version}-src";
   })
```
Comment (pbogdan) [inline pkgs/development/compilers/dmd/default.nix]: Unsure if this is proper way to deal with derivation with multiple sources.

File: pkgs/development/compilers/ldc/default.nix [hunk 6]
```diff
@@ -11,24 +11,28 @@ stdenv.mkDerivation rec {
...
     sha256 = "0fhcdfi7a00plwj27ysfyv783nhk0kspq7hawf6vbsl3s1nyvn8g";
+    name = "phobos-ldc-v${version}-src";
   })
   (fetchFromGitHub {
     owner = "ldc-developers";
     repo = "dmd-testsuite";
     rev = "ldc-v${version}";
     sha256 = "0dmdkp220gqhxjrmrjfkf0vsvylwfaj70hswavq4q3v4dg17pzmj";
+    name = "dmd-testsuite-ldc-v${version}-src";
   })
```
Comment (pbogdan) [inline pkgs/development/compilers/ldc/default.nix]: Unsure if this is proper way to deal with derivation with multiple sources.

**Review by orivej — APPROVED** [review]
Confirmed with `nox-review`. Thank you very much!

**Pull request merged**